UML editor/src/main/java/codecain/GraphicalUserInterface/Controller.java

```diff
@@ -912,52 +201,42 @@ private void saveBtn() throws IOException {
 
         FileChooser fileChooser = new FileChooser();
         fileChooser.setTitle("Save UML Diagram File");
-
-        // Set filter for JSON files
         fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
-        
-        // Set the initial directory to the current working directory
         fileChooser.setInitialDirectory(new File(System.getProperty("user.dir")));
 
         Window window = nodeContainer.getScene().getWindow();
 
         File file = fileChooser.showSaveDialog(window);
-        if(file != null) {
+        if (file != null) {
             SaveManager.saveToJSON(file.getAbsolutePath());
         }
     }
 
+    /**
+     * Handles loading a UML diagram from a JSON file.
+     *
+     * @throws IOException if there is an error during the load process.
+     */
     @FXML
     private void loadBtn() throws IOException {
-
         FileChooser fileChooser = new FileChooser();
         fileChooser.setTitle("Open UML Diagram File");
-
-        // Set filter for JSON files
         fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
-
-        // Set the initial directory to the current working directory
         fileChooser.setInitialDirectory(new File(System.getProperty("user.dir")));
 
-        // Get the main window for the file chooser
         Window window = nodeContainer.getScene().getWindow();
 
         File file = fileChooser.showOpenDialog(window);
-        if(file != null) {
+        if (file != null) {
             SaveManager.loadFromJSON(file.getAbsolutePath());
             populateGUIFromClassMap();
         }
     }
 
-
     /**
-     * Populates the GUI with `ClassNode` instances based on the current state of the UML class map.
-     * 
-     * This method clears the `nodeContainer` and iterates over all `UMLClassInfo` instances
-     * in the `UMLClass.classMap`. If a class has default coordinates (0, 0), it calculates
-     * and assigns a new non-overlapping position using {@link #calculateAndSetPosition(ClassNode, UMLClassInfo, Pane)}.
-     * Otherwise, it uses the existing coordinates stored in the `UMLClassInfo`.
-     * Each `ClassNode` is also set up with a click event for selection and added to the container.
+     * Populates the GUI with data from the backend UML class map.
+     * Clears the existing nodes in the GUI and repopulates them based on the
+     * backend state.
      */
     public void populateGUIFromClassMap() {
```

Comment:
the calulateAndSetPostion method needs to be implemented so we have NonStackable Boxes on creation.